### PR TITLE
iOS: Support GDPR (APSAnalytics 2.1.0), add missing native header

### PR DESCRIPTION
**Pull request by hansemannn**
**JIRA:** https://jira.appcelerator.org/browse/TIMOB-26004

Adding unit test once CR approved.

### CODE REVIEW

**Review by garymathews — DISMISSED** [review]
I think we need to agree on the API specification for Ti.Analytics optOut because this is varying per implementation:
```
Ti.Analytics.optedOut // instead of Ti.Analytics.isOptedOut
Ti.Analytics.setOptedOut(boolean)
Ti.Analytics.getOptedOut()
```

For consistency like our other Titanium APIs

For example, the Windows implementation uses `isOptOut`:
https://github.com/appcelerator/titanium_mobile_windows/pull/1238

And Android uses `optedOut`:
https://github.com/appcelerator/titanium_mobile/pull/10058

**Comment by hansemannn** [comment]
@garymathews @ypbnv @infosia I have updated the iOS-PR to add the property `optedOut` with it's getter `getOptedOut()` and setter `setOptedOut(true|false)`. I have also added the docs for iOS and Android - I think the "windows" platform tag needs to be added via the titanium_mobile_windows repo?

**Comment by infosia** [comment]
Updated Windows PR: https://github.com/appcelerator/titanium_mobile_windows/pull/1238 Updated API doc for Windows too.

**Comment by hansemannn** [comment]
@garymathews The PR was updated, can you re-review? I have also added @vijaysingh-axway for another CR.

File: iphone/Classes/AnalyticsModule.m [hunk 1]
```diff
@@ -105,6 +105,17 @@ - (void)filterEvents:(id)args
...
 
+- (void)setOptedOut:(id)optedOut
+{
+  ENSURE_TYPE(optedOut, NSNumber);
+  [[APSAnalytics sharedInstance] setOptedOut:[TiUtils boolValue:optedOut]];
+}
+
+- (void)optedOut
+{
+  @([[APSAnalytics sharedInstance] isOptedOut]);
```
Comment (vijaysingh-axway) [inline iphone/Classes/AnalyticsModule.m]: 1. Isn't it good if we make this method's return type to (NSNumber *) for more readability and return it.
2. Following test case crashing, if I close the app and run again. Can this not be a test case ?
`var win = Ti.UI.createWindow({
    backgroundColor: '#fff'
});

var btn = Ti.UI.createButton({
    title: 'Trigger'
});

btn.addEventListener('click', function() {
	Ti.Analytics.setOptedOut(true);
    });

win.add(btn);
win.open();
`

**Review by vijaysingh-axway — CHANGES_REQUESTED** [review]
@hansemannn Can you address the 2 points which I have mentioned. Thanks!

File: tests/ti.analytics.addontest.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+ * Please see the LICENSE included with this distribution for details.
+ */
+/* eslint-env mocha */
+/* global Ti */
+/* eslint no-unused-expressions: "off" */
+'use strict';
+var should = require('./utilities/assertions');
+
+describe('Titanium.Analytics', function () {
+	it.androidMissing('.optedOut', function () {
```
Comment (hansemannn) [inline tests/ti.analytics.addontest.js]: @garymathews @ypbnv Once Android is merged, we can update this one to work or all platforms.

**Review by vijaysingh-axway — APPROVED** [review]
CR passed.

**Pull request merged**